components/layout_thread_2020/lib.rs

```diff
@@ -186,89 +160,28 @@ pub struct LayoutThread {
-    busy: Arc<AtomicBool>,
-
     /// Debug options, copied from configuration to this `LayoutThread` in order
     /// to avoid having to constantly access the thread-safe global options.
     debug: DebugOptions,
 }
 
-impl LayoutThreadFactory for LayoutThread {
-    type Message = Msg;
-
-    /// Spawns a new layout thread.
-    fn create(
-        id: PipelineId,
-        top_level_browsing_context_id: TopLevelBrowsingContextId,
-        url: ServoUrl,
-        is_iframe: bool,
-        chan: (Sender<Msg>, Receiver<Msg>),
-        pipeline_port: IpcReceiver<LayoutControlMsg>,
-        background_hang_monitor_register: Box<dyn BackgroundHangMonitorRegister>,
-        constellation_chan: IpcSender<ConstellationMsg>,
-        script_chan: IpcSender<ConstellationControlMsg>,
-        image_cache: Arc<dyn ImageCache>,
-        font_cache_thread: FontCacheThread,
-        time_profiler_chan: profile_time::ProfilerChan,
-        mem_profiler_chan: profile_mem::ProfilerChan,
-        webrender_api_sender: WebrenderIpcSender,
-        paint_time_metrics: PaintTimeMetrics,
-        busy: Arc<AtomicBool>,
-        window_size: WindowSizeData,
-    ) {
-        thread::Builder::new()
-            .name(format!("Layout{}", id))
-            .spawn(move || {
-                thread_state::initialize(ThreadState::LAYOUT);
-
-                // In order to get accurate crash reports, we install the top-level bc id.
-                TopLevelBrowsingContextId::install(top_level_browsing_context_id);
-
-                {
-                    // Ensures layout thread is destroyed before we send shutdown message
-                    let sender = chan.0;
-
-                    let background_hang_monitor = background_hang_monitor_register
-                        .register_component(
-                            MonitoredComponentId(id, MonitoredComponentType::Layout),
-                            Duration::from_millis(1000),
-                            Duration::from_millis(5000),
-                            None,
-                        );
-
-                    let layout = LayoutThread::new(
-                        id,
-                        top_level_browsing_context_id,
-                        url,
-                        is_iframe,
-                        chan.1,
-                        pipeline_port,
-                        background_hang_monitor,
-                        constellation_chan,
-                        script_chan,
-                        image_cache,
-                        font_cache_thread,
-                        time_profiler_chan,
-                        mem_profiler_chan.clone(),
-                        webrender_api_sender,
-                        paint_time_metrics,
-                        busy,
-                        window_size,
-                    );
-
-                    let reporter_name = format!("layout-reporter-{}", id);
-                    mem_profiler_chan.run_with_memory_reporting(
```

Comment:
Will we still be able to profile memory and time with sufficient granularity (e.g script work vs layout work) after this change?

Reply:
I think we lose some granularity there, but:

1. The memory profile infrastructure is fairly abandoned right now.
2. This brings layout on par with script. If we need to break down object allocation per pipeline it would be better to do it in both layout and script in the same way.